src/stores/GuildMemberRoleStore.js

```diff
@@ -102,51 +49,98 @@ class GuildMemberRoleStore extends DataStore {
    * @readonly
    */
   get highest() {
-    return this.reduce((prev, role) => role.comparePositionTo(prev) > 0 ? role : prev, this.first());
+    return this._filtered.reduce((prev, role) => role.comparePositionTo(prev) > 0 ? role : prev, this.first());
   }
 
   /**
-   * Patches the roles for this store
-   * @param {Snowflake[]} roles The new roles
-   * @private
+   * Adds a role (or multiple roles) to the member.
+   * @param {RoleResolvable|RoleResolvable[]|Collection<Snowflake, Role>} roleOrRoles The role or roles to add
+   * @param {string} [reason] Reason for adding the role(s)
+   * @returns {Promise<GuildMember>}
    */
-  _patch(roles) {
-    this.clear();
-
-    const everyoneRole = this.guild.roles.get(this.guild.id);
-    if (everyoneRole) super.set(everyoneRole.id, everyoneRole);
+  async add(roleOrRoles, reason) {
+    if (roleOrRoles instanceof Collection || roleOrRoles instanceof Array) {
+      roleOrRoles = roleOrRoles.map(r => this.guild.roles.resolve(r));
+      if (roleOrRoles.includes(null)) {
+        return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
+          'Array or Collection of Roles or Snowflakes', true));
+      }
+      const newRoles = [...new Set(roleOrRoles.concat(...this.keys()))];
+      return this.set(newRoles, reason);
+    } else {
+      if (this.guild.roles.resolve(roleOrRoles) === null) {
+        return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
+          'Array or Collection of Roles or Snowflakes', true));
+      }
+      await this.client.api.guilds[this.guild.id].members[this.member.id].roles[roleOrRoles.id].put();
+      return this.member;
```

Comment:
`GuildMemberRoleStore#set` (above) calls `GuildMember#edit` which resolves with a cloned and locally patched guild member.
This inconsistently does simply resolve with the original and unpatched one.